gatsby-config.js

```diff
@@ -190,6 +190,7 @@ module.exports = {
         path: `./src/nav/`,
       },
     },
+    'gatsby-plugin-generate-doc-json',
```

Comment:
We might be able to leverage [`gatsby-plugin-json-output`](https://github.com/dominicfallows/gatsby-plugin-json-output) to generate these files. We use it below to generate a single JSON for the data dictionary (via `serializeFeed` option) but you can use it with the `serialize` option instead to generate a JSON file for each node.

Most of the logic you have would be the same, we just don't need to make a custom plugin!

Reply:
Yeah I first tried to do that, but the problem was that there is some specific log for it (specifically having to do with the path and the slug) which is why I opted for creating my own plugin, I'm also going to do some more transformations on the data so I still push for having it's own plugin

Reply:
```js
{
  resolve: 'gatsby-plugin-json-output',
  options: {
    siteUrl,
    graphQLQuery: `
      {
        ...your query goes here
      }
    `,
    serialize: results => results.data.allMdx.nodes.map((node) => {
      // your code goes here
    });
  }
}
```

Reply:
there's a json created for each path, rather than a json that has all the docs files (which would be generated if we use the serialize function. If we wanted to put all the docs files into one json i would advocate for using `gatsby-plugin-json-output` !

Reply:
Ah alright. I _think_ this library would still work for those edge cases, but I'm not going to complain about moving logic out of `gatsby-config.js` and into specialized files.

Reply:
OH you're suggesting using `gatsby-plugin-json-output` not the other one! Sorry I misread this

Reply:
I'm making some changes to this PR and doing some AST codemod things so it's easier for nr1 help xp to read, so let me push up those changes and then maybe you can determine whether it would still be feasible to put it in the config file